Mathlib/Data/Matrix/Notation.lean

```diff
@@ -0,0 +1,503 @@
+/-
+Copyright (c) 2020 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen, Eric Wieser
+
+! This file was ported from Lean 3 source module data.matrix.notation
+! leanprover-community/mathlib commit 3e068ece210655b7b9a9477c3aff38a492400aa1
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Matrix.Basic
+import Mathlib.Data.Fin.VecNotation
+import Mathlib.Tactic.FinCases
+import Mathlib.Tactic.ToExpr
+import Mathlib.Algebra.BigOperators.Fin
+import Qq
+
+/-!
+# Matrix and vector notation
+
+This file includes `simp` lemmas for applying operations in `Data.Matrix.Basic` to values built out
+of the matrix notation `![a, b] = vecCons a (vecCons b vecEmpty)` defined in
+`Data.Fin.VecNotation`.
+
+This also provides the new notation `!![a, b; c, d] = Matrix.of ![![a, b], ![c, d]]`.
+This notation also works for empty matrices; `!![,,,] : Matrix (Fin 0) (Fin 3)` and
+`!![;;;] : Matrix (Fin 3) (Fin 0)`.
+
+## Implementation notes
+
+The `simp` lemmas require that one of the arguments is of the form `vecCons _ _`.
+This ensures `simp` works with entries only when (some) entries are already given.
+In other words, this notation will only appear in the output of `simp` if it
+already appears in the input.
+
+## Notations
+
+This file provide notation `!![a, b; c, d]` for matrices, which corresponds to
+`Matrix.of ![![a, b], ![c, d]]`.
+Note that in lean 4 the pretty-printer will not show `!!` notation, instead showing the version
+with `of ![![...]]`.
```

Comment:
Lean didn't delaborate `![x, y, ...]` notations, so this was wrong.
So, I added unexpanders.

Reply:
It doesn't look like you pushed the unexpander?